rasterio/dtypes.py

```diff
@@ -6,6 +6,8 @@
 
 """
 import numpy as np
+from numpy.core._multiarray_umath import _discover_array_parameters
```

Comment:
If this was a public import, I think this would be a great path forward. However, it is importing a non-public function from a non-public module. I wouldn't trade long-term stability for the potential performance gains this may provide.

Reply:
Fair enough.